src/transformers/pipelines/text_to_audio.py

```diff
@@ -171,6 +192,10 @@ def _forward(self, model_inputs, **kwargs):
             # generate_kwargs get priority over forward_params
             forward_params.update(generate_kwargs)
 
+            # ensure dict output to facilitate postprocessing
+            if self.model.config.model_type not in ["bark", "musicgen"]:
+                forward_params.update({"return_dict_in_generate": True})
```

Comment:
added this as noted in the comment: having the dictionary makes it easier to detect whether calling `self.processor.decode(audio)` is needed. 

So fewer exceptions below, and only adding exceptions here for backwards compatibility 🙂